Library/SharedFunctions.swift

```diff
@@ -397,6 +397,88 @@ public func rounded(_ value: Double, places: Int) -> Double {
   return (value * divisor).rounded() / divisor
 }
 
+/**
+ An helper func that calculates  shipping total for base reward
+
+ - parameter project: The `Project` associated with a group of Rewards.
+ - parameter baseReward: The reward being evaluated
+
+ - returns: A `Double` of the shipping value. If the `Project` `Backing` object is nil,
+            and `baseReward` shipping is not enabled, the value is `0.0`
+ */
+public func getBaseRewardShippingTotal(project: Project, baseReward: Reward) -> Double {
+  guard baseReward.shipping.enabled, let backing = project.personalization.backing else { return 0.0 }
```

Comment:
So I just realized something while I was testing on your branch. On a new pledge this always return 0 because there is no `Backing`.  The desired behavior is to return the shipping amount designated on the Reward for a new pledge.

I tested this while doing a new pledge for the project `The Tarot Restless: Third Edition`. The shipping amount is $5 to the destination I chose but we are sending `checkout_shipping_amount`: 0 to Segment.